### ISSUE TITLE
out of range

### ISSUE THREAD

**Issue by tingfeng-key** [Lonami/grammers#228]
```rust
thread 'tokio-runtime-worker' panicked at C:\Users\x\Desktop\grammers\lib\grammers-mtsender\src\lib.rs:335:67:
range start index 1416 out of range for slice of length 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'tokio-runtime-worker' panicked at C:\Users\x\Desktop\grammers\lib\grammers-mtsender\src\lib.rs:335:67:
range start index 1416 out of range for slice of length 0
thread 'tokio-runtime-worker' panicked at C:\Users\x\Desktop\grammers\lib\grammers-mtsender\src\lib.rs:335:67:
range start index 1416 out of range for slice of length 0
thread 'tokio-runtime-worker' panicked at C:\Users\x\Desktop\grammers\lib\grammers-mtsender\src\lib.rs:335:67:
range start index 1416 out of range for slice of length 0
thread 'tokio-runtime-worker' panicked at C:\Users\x\Desktop\grammers\lib\grammers-mtsender\src\lib.rs:335:67:
range start index 1416 out of range for slice of length 0
```

![image](https://github.com/Lonami/grammers/assets/13976521/e26344be-cd86-4802-a943-8dd051df7cc0)

**Comment by Lonami** [comment]
What steps are needed to reproduce the issue?

**Comment by tingfeng-key** [comment]
I don't know how to reproduce it. It just appears very suddenly. My program runs locally for a long time. I have encountered it several times recently, sometimes by accident.

**Comment by Lonami** [comment]
Turns out there was a pretty serious bug with how resetting occurred, which should be fixed by 9c8d738f3622bcf9935ea13dfbd80418ed7addca, so assuming this was fixed, I'll close the issue.
